cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplications.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.capability;
+
+import io.cdap.cdap.proto.id.ApplicationId;
+
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Class for holding search results of a search for applications annotated with capabilities
+ */
+public class CapabilityApplications {
```

Comment:
Generally class name with `s` at the end is for helper/util class. Also, this class seems quite generic that it is not specific for capability. Seems like you can have something like:

```
public class EntityResult<T> {
  private final Set<T> entities;
  ...
}
--------
EntityResult<ApplicationId> getApplications(...)
```

Reply:
changed to generic class